promunifi/clients.go

```diff
@@ -0,0 +1,126 @@
+package promunifi
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"golift.io/unifi"
+)
+
+type uclient struct {
+	Anomalies         *prometheus.Desc
+	BytesR            *prometheus.Desc
+	CCQ               *prometheus.Desc
+	Noise             *prometheus.Desc
+	RoamCount         *prometheus.Desc
+	RSSI              *prometheus.Desc
+	RxBytes           *prometheus.Desc
+	RxBytesR          *prometheus.Desc
+	RxPackets         *prometheus.Desc
+	RxRate            *prometheus.Desc
+	Signal            *prometheus.Desc
+	TxBytes           *prometheus.Desc
+	TxBytesR          *prometheus.Desc
+	TxPackets         *prometheus.Desc
+	TxPower           *prometheus.Desc
+	TxRate            *prometheus.Desc
+	Uptime            *prometheus.Desc
+	WifiTxAttempts    *prometheus.Desc
+	WiredRxBytes      *prometheus.Desc
+	WiredRxBytesR     *prometheus.Desc
+	WiredRxPackets    *prometheus.Desc
+	WiredTxBytes      *prometheus.Desc
+	WiredTxBytesR     *prometheus.Desc
+	WiredTxPackets    *prometheus.Desc
+	DpiStatsApp       *prometheus.Desc
+	DpiStatsCat       *prometheus.Desc
+	DpiStatsRxBytes   *prometheus.Desc
+	DpiStatsRxPackets *prometheus.Desc
+	DpiStatsTxBytes   *prometheus.Desc
+	DpiStatsTxPackets *prometheus.Desc
+}
+
+func descClient(ns string) *uclient {
+	if ns += "_client_"; ns == "_client_" {
+		ns = "client_"
+	}
+
+	labels := []string{"id", "mac", "user_id", "site_id", "site_name",
+		"network_id", "ap_mac", "gw_mac", "sw_mac", "ap_name", "gw_name",
+		"sw_name", "radio_name", "radio", "radio_proto", "name", "channel",
+		"vlan", "ip", "essid", "bssid", "radio_desc"}
+
+	return &uclient{
+		Anomalies:         prometheus.NewDesc(ns+"anomalies_total", "Client Anomalies", labels, nil),
+		BytesR:            prometheus.NewDesc(ns+"bytes_rate", "Client Data Rate", labels, nil),
+		CCQ:               prometheus.NewDesc(ns+"ccq", "Client Connection Quality", labels, nil),
+		Noise:             prometheus.NewDesc(ns+"noise", "Client AP Noise", labels, nil),
+		RoamCount:         prometheus.NewDesc(ns+"roam_total", "Client Roam Counter", labels, nil),
+		RSSI:              prometheus.NewDesc(ns+"rssi", "Client RSSI", labels, nil),
+		RxBytes:           prometheus.NewDesc(ns+"bytes_rx_total", "Client Receive Bytes", labels, nil),
+		RxBytesR:          prometheus.NewDesc(ns+"bytes_rx_rate", "Client Receive Data Rate", labels, nil),
+		RxPackets:         prometheus.NewDesc(ns+"packets_rx_total", "Client Receive Packets", labels, nil),
```

Comment:
We like to use full names, rather than abbreviations when possible, as it's more descriptive. Also, we like to put the unit towards the end, to make the metric name more readable.

```suggestion
		RxPackets:         prometheus.NewDesc(ns+"receive_packets_total", "Client Receive Packets", labels, nil),
```

Reply:
if this was a rate instead of a total, would you put `_rate` and the end or `rate_` at the beginning?

Reply:
Sorry to ask this again........

Reply:
No worries, it's a convention to make reading metric names easy. If it's a gauge, it would simply leave off the `_total`, ie `receive_packets`. This is because the "unit" in this data is packets.